pkg/migration/sqlmigrations/migrations_test.go

```diff
@@ -441,12 +447,17 @@ func TestCreateSystemTable(t *testing.T) {
 	settingsDescVal := sqlbase.WrapDescriptor(&settingsDesc)
 
 	// Start up a test server without running the system.jobs migration.
-	backwardCompatibleMigrations = []migrationDescriptor{
-		{
-			name:   "create system.settings table",
-			workFn: createSettingsTable,
-		},
+	// remove the view update migration so we can test its effects.
+	newMigrations := make([]migrationDescriptor, 0, len(backwardCompatibleMigrations))
+	for _, m := range backwardCompatibleMigrations {
+		if strings.HasPrefix(m.name, "create system.jobs") {
+			continue
+		}
+		newMigrations = append(newMigrations, m)
```

Comment:
This hunk is what makes the test flaky. (Really, it should make the test consistently fail.) It's not valid to slice out a migration like this, I'm afraid—`AdditionalInitialDescriptors` takes some shortcuts because it knows all migrations will be run in sequence. 

Specifically, if you don't run the jobs migration, `AdditionalInitialDescriptors` doesn't see the `newRanges: 1` on the jobs migration. This seems logical—the table hasn't been created! But the just-added web sessions table has ID 19, and you've left that migration in the list. The split queue always splits at every table boundary, up to the maximum present table ID, even if the tables in-between are missing. So when table 19 is created, table ID 15 (the jobs table) will get a range, even though the table doesn't exist. 

It's all rather messy. The old code approach of including only the `system.settings` migration because it's required for server startup seems palatable enough, though. We could always break out a `migrationsRequiredEvenToStartUpATestServer` variable. With a better name, of course.